Mathlib/Data/Matroid/Rank/ENat.lean

```diff
@@ -298,6 +261,170 @@ lemma eRk_inter_add_eRk_union_le (M : Matroid α) (X Y : Set α) :
+
+lemma eRk_union_le_eRk_add_eRk (M : Matroid α) (X Y : Set α) : M.eRk (X ∪ Y) ≤ M.eRk X + M.eRk Y :=
+  le_add_self.trans (M.eRk_submod X Y)
+
+lemma eRk_eq_eRk_union_eRk_le_zero (X : Set α) (hY : M.eRk Y ≤ 0) : M.eRk (X ∪ Y) = M.eRk X :=
+  (((M.eRk_union_le_eRk_add_eRk X Y).trans (add_le_add_left hY _)).trans_eq (add_zero _)).antisymm
+    (M.eRk_mono subset_union_left)
+
+lemma eRk_eq_eRk_diff_eRk_le_zero (X : Set α) (hY : M.eRk Y ≤ 0) : M.eRk (X \ Y) = M.eRk X := by
+  rw [← eRk_eq_eRk_union_eRk_le_zero (X \ Y) hY, diff_union_self, eRk_eq_eRk_union_eRk_le_zero _ hY]
+
+lemma eRk_le_eRk_inter_add_eRk_diff (M : Matroid α) (X Y : Set α) :
+    M.eRk X ≤ M.eRk (X ∩ Y) + M.eRk (X \ Y) := by
+  nth_rw 1 [← inter_union_diff X Y]; apply eRk_union_le_eRk_add_eRk
+
+lemma eRk_le_eRk_add_eRk_diff (M : Matroid α) (h : Y ⊆ X) :
+    M.eRk X ≤ M.eRk Y + M.eRk (X \ Y) := by
+  nth_rw 1 [← union_diff_cancel h]; apply eRk_union_le_eRk_add_eRk
+
+lemma eRk_union_le_encard_add_eRk (M : Matroid α) (X Y : Set α) :
+    M.eRk (X ∪ Y) ≤ X.encard + M.eRk Y :=
+  (M.eRk_union_le_eRk_add_eRk X Y).trans <| add_le_add_right (M.eRk_le_encard _) _
+
+lemma eRk_union_le_eRk_add_encard (M : Matroid α) (X Y : Set α) :
+    M.eRk (X ∪ Y) ≤ M.eRk X + Y.encard :=
+  (M.eRk_union_le_eRk_add_eRk X Y).trans <| add_le_add_left (M.eRk_le_encard _) _
+
+lemma eRank_le_encard_add_eRk_compl (M : Matroid α) (X : Set α) :
+    M.eRank ≤ X.encard + M.eRk (M.E \ X) :=
+  le_trans (by rw [← eRk_inter_ground, eRank_def, union_diff_self,
+    union_inter_cancel_right]) (M.eRk_union_le_encard_add_eRk X (M.E \ X))
+
 end Basic
 
+/-! ### Finiteness -/
+
+@[simp]
+lemma eRank_ne_top_iff (M : Matroid α) : M.eRank ≠ ⊤ ↔ M.RankFinite := by
+  obtain ⟨B, hB⟩ := M.exists_isBase
+  rw [← hB.encard_eq_eRank, encard_ne_top_iff]
+  exact ⟨fun h ↦ hB.rankFinite_of_finite h, fun h ↦ hB.finite⟩
+
+@[deprecated (since := "2025-04-13")] alias rankFinite_iff_eRk_ne_top := eRank_ne_top_iff
+
+lemma eRank_eq_top_iff (M : Matroid α) : M.eRank = ⊤ ↔ M.RankInfinite := by
+  rw [← not_rankFinite_iff, ← eRank_ne_top_iff, not_not]
+
+@[deprecated (since := "2025-04-13")] alias rankInfinite_iff_eRk_eq_top := eRank_eq_top_iff
+
+@[simp]
+lemma eRank_eq_top [RankInfinite M] : M.eRank = ⊤ :=
+  (eRank_eq_top_iff _).2 <| by assumption
+
+@[simp]
+lemma eRk_eq_top_iff : M.eRk X = ⊤ ↔ ¬ M.IsRkFinite X := by
+  obtain ⟨I, hI⟩ := M.exists_isBasis' X
+  rw [hI.eRk_eq_encard, encard_eq_top_iff, ← hI.finite_iff_isRkFinite, Set.Infinite]
+
+lemma eRk_ne_top_iff : M.eRk X ≠ ⊤ ↔ M.IsRkFinite X := by
+  simp
+
+@[simp]
+lemma eRk_lt_top_iff : M.eRk X < ⊤ ↔ M.IsRkFinite X := by
+  rw [lt_top_iff_ne_top, eRk_ne_top_iff]
+
+lemma IsRkFinite.eRk_lt_top (h : M.IsRkFinite X) : M.eRk X < ⊤ :=
+  eRk_lt_top_iff.2 h
+
+@[deprecated (since := "2025-04-13")] alias eRk_lt_top_of_finite := IsRkFinite.eRk_lt_top
+
+lemma IsRkFinite.eRk_ne_top (h : M.IsRkFinite X) : M.eRk X ≠ ⊤ :=
+  h.eRk_lt_top.ne
```

Comment:
The usual setup would be to have only the `lt_top` one named, since this one can be derived easily by `.ne`. (Whereas the `iff` versions you have are completely fine to keep)